botocore/httpsession.py

```diff
@@ -224,6 +247,26 @@ def _setup_ssl_cert(self, conn, url, verify):
             conn.cert_reqs = 'CERT_NONE'
             conn.ca_certs = None
 
+    def _setup_proxy_ssl_context(self, proxies_settings):
+        proxy_ca_bundle = proxies_settings.get('proxy_ca_bundle')
+        proxy_cert = proxies_settings.get('proxy_client_cert')
+        if proxy_ca_bundle is None and proxy_cert is None:
+            return None
+
+        context = self._get_ssl_context()
+        try:
```

Comment:
Does urllib3 not have a similar `cert_file` and `key_file` kwarg for the proxies and that's why we're doing this ourselves?

Reply:
Right, they only have that on the PoolManager class. They expect a pre-constructed `SSLContext` for the ProxyManager.